src/app/components/search/search.component.spec.ts

```diff
@@ -52,11 +57,38 @@ describe('SearchComponent', () => {
     expect(component.onSubmit).toHaveBeenCalled();
   });
 
-  it('should clear search text', () => {
-    const search = component.form.controls['searchText'];
+  it('should call submit function when search button is clicked and fields are filled', () => {
+    spyOn(component, 'onSubmit').and.callThrough();
+
+    const search = component.form.controls['case_number'];
+    search.setValue('1');
+    const ch = component.form.controls['courthouse'];
+    ch.setValue('Reading');
+    const cr = component.form.controls['courtroom'];
+    cr.setValue('2');
+    const jn = component.form.controls['judge_name'];
+    jn.setValue('Judy');
+    const dn = component.form.controls['defendant_name'];
+    dn.setValue('Dave');
+    const dt = component.form.controls['date_to'];
+    dt.setValue('18/09/2023');
+    const df = component.form.controls['date_from'];
+    df.setValue('19/09/2023');
+    const kw = component.form.controls['event_text_contains'];
+    kw.setValue('Keywords');
+
+    fixture.debugElement.query(By.css('form')).triggerEventHandler('ngSubmit', null);
+    fixture.detectChanges();
+
+    expect(component.onSubmit).toHaveBeenCalled();
```

Comment:
Can we also check that the case service has been called with the appropriate parameters?

Reply:
Also the error scenario?

Reply:
I have raised separate ticket (DMP-774) to test error responses as i think we'll need a mock controller to return expected responses properly. it is worth doing this in a separate ticket iMO as we can create something adaptable and modular that can be used for other services going forward.